tritium-metrics/src/main/java/com/palantir/tritium/metrics/executor/TaggedMetricsExecutorService.java

```diff
@@ -0,0 +1,199 @@
+        submitted.mark(tasks.size());
+        Collection<TaggedMetricsCallable<T>> instrumented = instrument(tasks);
+        return delegate.invokeAny(instrumented);
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+            throws ExecutionException, InterruptedException, TimeoutException {
+        submitted.mark(tasks.size());
+        Collection<TaggedMetricsCallable<T>> instrumented = instrument(tasks);
+        return delegate.invokeAny(instrumented, timeout, unit);
+    }
+
+    private <T> Collection<TaggedMetricsCallable<T>> instrument(Collection<? extends Callable<T>> tasks) {
+        List<TaggedMetricsCallable<T>> instrumented = new ArrayList<>(tasks.size());
+        for (Callable<T> task : tasks) {
+            instrumented.add(new TaggedMetricsCallable<>(task));
+        }
+        return instrumented;
+    }
+
+    @Override
+    public void shutdown() {
+        delegate.shutdown();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+        return delegate.shutdownNow();
+    }
+
+    @Override
+    public boolean isShutdown() {
+        return delegate.isShutdown();
+    }
+
+    @Override
+    public boolean isTerminated() {
+        return delegate.isTerminated();
+    }
+
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+        return delegate.awaitTermination(timeout, unit);
+    }
+
+    private class TaggedMetricsRunnable implements Runnable {
+
+        private final Runnable task;
+
+        private final Timer.Context queuedContext = queuedDuration.time();
+
+        TaggedMetricsRunnable(Runnable task) {
+            this.task = task;
+        }
+
+        @Override
+        public void run() {
+            queuedContext.stop();
+            running.inc();
+            try (Timer.Context ignored = duration.time()) {
+                task.run();
+            } finally {
+                running.dec();
+                completed.mark();
+            }
+        }
+    }
+
+    private class TaggedMetricsCallable<T> implements Callable<T> {
+
+        private final Callable<T> task;
+
+        private final Timer.Context queuedContext = queuedDuration.time();
+
+        TaggedMetricsCallable(Callable<T> task) {
+            this.task = task;
+        }
+
+        @Override
+        public T call() throws Exception {
+            queuedContext.stop();
+            running.inc();
+            try (Timer.Context ignored = duration.time()) {
+                return task.call();
+            } finally {
+                running.dec();
+                completed.mark();
+            }
+        }
+    }
+
+    private static MetricName createMetricName(String metricName, Map<String, String> safeTags) {
+        return MetricName.builder()
+                .safeName(MetricRegistry.name("com/palantir/tritium/metrics/executor", metricName))
```

Comment:
probably don't want the long slashed name

```suggestion
                .safeName(MetricRegistry.name("executor", metricName))
```